nginx/setup.sh

```diff
@@ -1,82 +1,49 @@
-	exit 1
-fi
-
-echo "Checking AWS credentials are valid"
-if [ -z "$1" ]; then
-	PROFILE=""
-else
-	PROFILE="--profile ${1}"
-fi
-
-aws ${PROFILE} s3 ls ${S3_BUCKET} 1>/dev/null 2>&1
-if [[ $? -gt 0 ]]; then
-	echo "ERROR: You do not have access to the Identity AWS account. Re-run with ${0} <profile name> to use a different profile."
-	exit 2
-fi
+CERT_DIRECTORY=$HOME/.gu/mkcert
+KEY_FILE=${CERT_DIRECTORY}/${DOMAIN}.key
+CERT_FILE=${CERT_DIRECTORY}/${DOMAIN}.crt
 
-echo "Checking Nginx sites-enabled directory exists"
-ls "${nginxHome}/sites-enabled" 1>/dev/null 2>&1
-if [[ $? -gt 0 ]]; then
-	echo "ERROR: Missing ${nginxHome}/sites-enabled"
-    echo "Create sites-enabled directory and make sure it is included in your nginx.conf (usually in ${nginxHome}/nginx.conf):
-            http {
-                include       mime.types;
-                default_type  application/octet-stream;
-                # THIS IS WHAT YOU MUST ADD
-                include sites-enabled/*;
-            #..."
-	exit 1
-fi
-
-function install_ssh_certificate() {
-	KEY_NAME="${1}.key"
-	CRT_NAME="${1}.crt"
-	echo "Downloading SSL certificate $CRT_NAME from $S3_BUCKET"
-	aws ${PROFILE} s3 cp "${S3_BUCKET}${KEY_NAME}" ${DIR} 1>/dev/null
-	echo "Downloading SSL key $KEY_NAME from $S3_BUCKET"
-	aws ${PROFILE} s3 cp "${S3_BUCKET}${CRT_NAME}" ${DIR} 1>/dev/null
+# mkcert requires the JAVA_HOME envvar to be set to add the generated CA to Java's trust store
+# see https://github.com/FiloSottile/mkcert#supported-root-stores
+export JAVA_HOME=$(/usr/libexec/java_home)
 
-	echo "Linking SSL certs/keys to $nginxHome"
-	sudo ln -fs "${DIR}/${CRT_NAME}" "${nginxHome}/${CRT_NAME}"
-	sudo ln -fs "${DIR}/${KEY_NAME}" "${nginxHome}/${KEY_NAME}"
-}
+mkcert -install
 
-function install_ssh_certificate_in_jdk_ca() {
-    echo "Importing SSL certificate into Java keystore ${jdkHome}/jre/lib/security/cacerts"
-	sudo "${jdkHome}/bin/keytool" -import -alias ${1} -keystore "${jdkHome}/jre/lib/security/cacerts" -file "${1}.crt" -storepass "changeit" -noprompt
-}
+echo -e "🔐 Creating certificate for: ${YELLOW}$@${NC}"
+mkdir -p ${CERT_DIRECTORY}
+mkcert -key-file=${KEY_FILE} -cert-file=${CERT_FILE} ${DOMAIN}
 
-function install_nginx_configuration() {
-    echo "Linking Nginx configuration ${1} to $nginxHome/sites-enabled"
-    sudo ln -fs "$DIR/${1}" "$nginxHome/sites-enabled/${1}"
-}
+echo -e "🔗 Symlinking the certificate for nginx at ${NGINX_HOME}"
+ln -sf ${KEY_FILE} ${NGINX_HOME}/${DOMAIN}.key
+ln -sf ${CERT_FILE} ${NGINX_HOME}/${DOMAIN}.crt
 
-install_ssh_certificate ${SSL_CERT_NAME}
-install_ssh_certificate_in_jdk_ca ${SSL_CERT_NAME}
-install_nginx_configuration ${NGINX_SITE_CONF}
+echo -e "🔗 Symlinking nginx config file"
+ln -sf ${SOURCE_DIR}/frontend.conf ${NGINX_HOME}/servers/frontend.conf
 
-echo "Restarting Nginx"
-sudo nginx -s stop
+echo -e "🚀 ${YELLOW}Restarting nginx, Requires sudo - enter password when prompted.${NC}"
+if pgrep 'nginx' > /dev/null; then
```

Comment:
@nicl we're now restarting nginx more gracefully - if running, stop, finally start.